custom_components/tryfi/pytryfi/fiPet.py

```diff
@@ -160,6 +190,93 @@ def updateAllDetails(self, sessionId: requests.Session):
         # TODO: Support weekly
         self._dailySleep, self._dailyNap = self._extractSleep(petJson['dailySleepStat'])
         self._monthlySleep, self._monthlyNap = self._extractSleep(petJson['monthlySleepStat'])
+        # Try to fetch behavior data for Series 3+ collars
+        try:
+            self.updateBehaviorStats(sessionId)
+        except Exception:
+            # Behavior stats may not be available for older collars
+            pass
+
+    # Update behavior stats for Series 3+ collars
+    def updateBehaviorStats(self, sessionId: requests.Session):
+        """Update behavior statistics for Series 3+ collars."""
+        try:
+            healthTrendsJSON = query.getPetHealthTrends(sessionId, self.petId, 'DAY')
+            behavior_trends = healthTrendsJSON.get('behaviorTrends', [])
+            self.setBehaviorStatsFromTrends(behavior_trends)
+            return True
+        except Exception as e:
+            LOGGER.debug(f"Could not update behavior stats for Pet {self.name}. This may be an older collar model.\n{e}")
+            return False
```

Comment:
The updateAllInfo also has a try catch, making this handle redundant. I think it would be better to remove this, let it throw, then catch and log in updateAllInfo